lib/util/trip/ride_card.dart

```diff
@@ -1,84 +1,195 @@
+      super.trip = _ride;
+    });
+    loadRide();
+  }
+
+  @override
+  void didUpdateWidget(RideCard oldWidget) {
+    if (!trip!.equals(widget.trip)) {
+      loadRide();
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  Future<void> loadRide() async {
+    Ride trip = widget.trip;
+    Map<String, dynamic> data =
+        await SupabaseManager.supabaseClient.from('drives').select(_driveQuery).eq('id', trip.driveId).single();
+    trip.drive = Drive.fromJson(data);
+    if (mounted) {
+      setState(() {
+        _ride = trip;
+        _driver = trip.drive!.driver!;
+        super.trip = _ride;
+        _fullyLoaded = true;
+      });
+    }
+  }
+
+  @override
+  Widget buildTopRight() {
+    return Text(" ${_ride!.price}€");
+  }
+
+  @override
+  Widget buildBottomLeft() {
+    return !_fullyLoaded
+        ? const Center(
+            child: SizedBox(
+            height: 64,
+            width: 72,
+          ))
+        : Padding(
+            padding: const EdgeInsets.fromLTRB(16, 8, 16, 16),
+            child: ProfileWidget(_driver!),
+          );
+  }
+
+  @override
+  Widget buildBottomRight() {
+    return !_fullyLoaded
+        ? const Center(
+            child: SizedBox(
+            height: 44,
+            width: 52,
+          ))
+        : Padding(
+            padding: const EdgeInsets.fromLTRB(16, 8, 16, 16),
+            child: CustomRatingBarIndicator(
+                rating: AggregateReview.fromReviews(_ride!.drive!.driver!.reviewsReceived!).rating,
+                size: CustomRatingBarSize.medium),
+          );
+  }
+
+  @override
+  Widget buildRightSide() {
+    if (!_fullyLoaded) {
+      return const Center(
+          child: SizedBox(
+        height: 24,
+        width: 24,
+      ));
+    } else {
+      List<ProfileFeature> profileFeatures = _driver!.profileFeatures!;
+      List<Icon> featureicons = <Icon>[];
+      for (int i = 0; i < min(profileFeatures.length, 3); i++) {
+        featureicons.add(profileFeatures[i].feature.getIcon(context));
+      }
+      return Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: featureicons,
+      );
+    }
+  }
+
+  Color pickBannerColor() {
+    switch (_ride!.status) {
+      case RideStatus.preview:
+        return Theme.of(context).cardColor;
+      case RideStatus.pending:
+        return Theme.of(context).disabledColor;
+      case RideStatus.approved:
+        return Theme.of(context).own().success;
+      case RideStatus.rejected:
+        return Theme.of(context).colorScheme.error;
+      case RideStatus.cancelledByDriver:
+        return Theme.of(context).colorScheme.error;
+      case RideStatus.cancelledByRider:
+        return Theme.of(context).colorScheme.error;
+      case RideStatus.withdrawnByRider:
+        return Theme.of(context).cardColor;
+    }
+  }
+
+  BoxDecoration pickDecoration() {
+    if (_ride!.status.isCancelled() || _ride!.status == RideStatus.rejected) {
+      return BoxDecoration(
```

Comment:
this looks okay-ish in light mode, but have tried out dark mode? No thank you. I think we need to find a better way here to grey out the card